lib/Sema/CSSimplify.cpp

```diff
@@ -1908,6 +1908,12 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
     return getTypeMatchFailure(locator);
   }
 
+  if (Options.contains(ConstraintSystemFlags::UseClangFunctionTypes)) {
```

Comment:
This is incorrect; it out-right forbids conversion from an `@convention(c)` function type to an `@convention(swift)` function type when `UseClangFunctionTypes`. However, code that was working earlier in this sense should continue to work... I also need to add tests for conversions between `@convention(c, cType: "...")` -> `@convention(swift)`.

Reply:
Fixed by moving the checks inside `matchFunctionRepresentation`, since, in some sense, the `cType: ` is really part of the convention/representation.